yaml.go

```diff
@@ -129,8 +129,8 @@ func (dec *Decoder) Decode(v interface{}) (err error) {
 		out = out.Elem()
 	}
 	d.unmarshal(node, out)
-	if len(d.terrors) > 0 {
-		return &TypeError{d.terrors}
+	if len(d.terrors) > 0 || len(d.strictErrors) > 0 {
```

Comment:
not sure if this is now common enough to have its own helper func attached to the decoder, something like `if err := d.checkErr(); err != nil { return err }` ... it would add some overhead due to one extra fn call so not sure if it's worth it.